### ISSUE TITLE
ui: "Delete secret" makes the entire sub-folder in the mount point disappear

### ISSUE THREAD

**Issue by lrstanley** [hashicorp/vault#5960]
When deleting a secret in the UI, it seems to be removing the entire "folder"/sub-path inside of the given mount point from the folder list.

1. Login to the web UI
2. Under a given mount, e.g. `foo/`, create two secrets: `foo/bar/baz key=value` and `foo/bar/bang key=value2`
3. Once saved, click "Delete secret" on `foo/bar/bang`.

**Expected behavior:**
The UI should drop you back into the last parent path/folder that still exists (since the parent may not exist if that was the only key in that path/folder). In this case, `foo/bar/`.

**What happens:**
The UI brings you back to the mount point list page (`foo/`), and the sub-folder which still contains secrets (`foo/bar/`) then disappears from the UI (this made me think it completely deleted `foo/bar/` -- but it didn't -- that gave me a good scare hah). A refresh of the page shows the `bar/` folder in `foo/` again.

For many sub-paths, it occurs as well. E.g. `foo/1/2/3/4 key=value`, where 1, 2, and 3 have other content within them, `1` will still disappear from my testing.

Sorry if my description is a bit confusing.. hard to explain.

**Environment:**
* Vault Server Version: v1.0.0
* Vault CLI Version: n/a
* Server Operating System/Architecture: n/a

**Comment by khicks** [comment]
I get the worry about not being dropped to parent path where the deleted secret once was, and I'd definitely be startled if the whole path disappeared from the engine root. However, I can't seem to replicate  the temporary disappearance of existing paths of the deleted secret.

It would still be nice to be dropped back to the parent path of the deleted secret, if it still exists. I do find it a bit annoying to have to navigate to where I once was to work with the remaining secrets.

**Comment by lrstanley** [comment]
Will try and replicate today using a dev vault instance with brand new mounts. Will see if I can GIF it.

**Comment by lrstanley** [comment]
Was able to replicate using a brand new v1.0.0 environment:

![vault5960](https://user-images.githubusercontent.com/1847365/50162379-35196d00-02ac-11e9-83b4-c967aa5a87ee.gif)

In this GIF, I spun up a v1.0.0 instance of Vault on windows (using `vault.exe server -dev`).

1. I'm logging in using the root token.
2. I already pre-created a `kv/` route that is `kv` version 1 (not 2!)
3. I create a key named `key` at `kv/1/2/3/4` with the value `value`.
4. I create a key named `key` at `kv/1/2/3/4a` with the value `value`.
5. I deleted the key at: `kv/1/2/3/4a` using the right ellipsis dropdown menu.
6. This dropped me back to `kv/`, where `1/2/` still shows up, but `3/` is no longer there/it says there are no secrets matching that path (even though `kv/1/2/3/4` still exists).
7. I tried clearing the search box to see if it was related to that -- nope.
8. I reload the page and re-login using the root token. Now the `kv/1/2/3/4` key named `key` exists.

This test was slightly different than what I described in my original issue, due to how many sub folders there were in my test, but otherwise it's the same.
